### PR TITLE
Add multiple input features

### PR DESCRIPTION
**Pull request by devoxel**
This also does a few other things:

- ensure pep8 standard is met
- reoganize input module

### CODE REVIEW

File: passflip/arguments.py [hunk 1]
```diff
@@ -15,10 +15,15 @@ def init_arguments(self):
             help="prompt for password & salt twice."
         )
         self.add_argument(
+            "-m",
+            "--multiple",
+            action="store_true",
+            help="prompt forever until input is blank"
+        )
+        self.add_argument(
```
Comment (byxor) [inline passflip/arguments.py]: Looks good. I'm quite happy with how I chose to put the command line arguments in one place. It worked out well for new ones being added.

File: passflip/input.py [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 
-def prompt_password(again=False):
-    return prompt("enter password", again)
-
-
-def prompt_salt(again=False):
-    return prompt("enter salt", again)
-
+def prompt_salt(validate):
+    return prompt("enter salt", validate)
```
Comment (byxor) [inline passflip/input.py]: `prompt_password` and `prompt_salt` look better now.

File: passflip/input.py [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
+def prompt(message, validate):
+    first_input = getpass(message + ': ')
+    if len(first_input) == 0:
+        return None
+    if not validate:
+        return first_input
+    second_input = getpass(message + 'again: ')
+    if first_input != second_input:
+        first_input = None
+    return first_input
```
Comment (byxor) [inline passflip/input.py]: I know I wrote it, but I've forgotten what the return value of this function is meant to be.

Could you rename the function to make it clear what it returns?

File: test/input_test.py [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
+    def setUp(self):
+        self.validate = False;
 
-    def test_that_again_is_added_to_message(self):
-        self.given_the_message("Foo")
-        self.when_converted_to_a_prompt_message_with_again()
-        self.then_the_result_will_be("Foo again: ")
+    def test_that_user_can_input_password(self):
+        expected = "foocar"
+        self.when_user_is_prompted_for_password_and_they_input(expected)
```
Comment (byxor) [inline test/input_test.py]: These tests are fine, but they're a little confusing.

What's "the result" meant to be from prompting a user for a password?

File: test/input_test.py [hunk 5]
```diff
@@ -1,28 +1,35 @@
...
+        self.then_the_result_will_be(expected)
 
-    def given_the_message(self, message):
-        self.message = message
+    def test_that_user_can_input_salt(self):
+        expected = "barnod"
+        self.when_user_is_prompted_for_password_and_they_input(expected)
```
Comment (byxor) [inline test/input_test.py]: This is calling the `prompt_password` function, not the `prompt_salt` function.

File: test/input_test.py [hunk 6]
```diff
@@ -1,28 +1,35 @@
...
-    def when_converted_to_a_prompt_message(self):
-        self.prompt_message = create_prompt_message(self.message)
+    def when_user_is_prompted_for_password_and_they_input(self, expected):
+        with mock.patch.object(passflip.input, "getpass", create=True, return_value=expected) as m:
+            self.input = passflip.input.prompt_password(self.validate)
+            m.assert_called_once_with('enter password: ')
 
-    def when_converted_to_a_prompt_message_with_again(self):
-        self.prompt_message = create_prompt_message(self.message, again=True)
+    def when_user_is_prompted_for_password_and_they_input(self, expected):
```
Comment (byxor) [inline test/input_test.py]: should be
`def when_user_is_prompted_for_salt_and_they_input(self, expected):`

File: test/input_test.py [hunk 7]
```diff
@@ -1,28 +1,35 @@
 from unittest import TestCase
-from passflip.input import create_prompt_message
+import passflip.input
 
+try:
+    import mock
+except ImportError:
+    from unittest import mock
 
 class InputPromptMessageTest(TestCase):
```
Comment (byxor) [inline test/input_test.py]: These tests have changed quite a bit and the class name no longer reflects their intent. Maybe `PromptTest` or something would suit better.

**Review by byxor — CHANGES_REQUESTED** [review]
This looks really good. You've cleaned up quite a lot of things. I've commented out some minor mistakes and potential improvements.

I'm being really picky with the codebase because it's so small that it's easy to keep clean. My pickiness isn't a reflection of bad PR quality; I like what I see.

File: passflip/arguments.py (hunk 1 above)
Comment (devoxel) [inline passflip/arguments.py]: Re-organizing it would almost be the exact same thing except without the inheritance stuff. I have never seen that anywhere.

File: passflip/input.py (hunk 3 above)
Comment (devoxel) [inline passflip/input.py]: I'm not sure what to name it. It prompts for an input and can validate it. `get_input` sounds silly, `input_prompt` maybe? You have an idea?

File: passflip/arguments.py (hunk 1 above)
Comment (byxor) [inline passflip/arguments.py]: Oh right, we're talking about the inheritance now? Yeah, nobody's disagreeing with you.

File: passflip/input.py (hunk 3 above)
Comment (byxor) [inline passflip/input.py]: What does it return? Sometimes it returns the first thing typed in, sometimes it returns `None`. I think the `first_input = None` rather than `return None` is confusing me more.

Comment (devoxel) [inline passflip/input.py]: yeah okay I like the idea of the exception

**Comment by devoxel** [comment]
Okay I've fixed up what you were talking about. However, I don't think my tests are up to **Brandon**™ test  standards. You can fix them yourself probably? Also any recommended reading on writing that style of test?

**Comment by iandioch** [comment]
Any updates on this..?

**Comment by byxor** [comment]
I'll read over it again when I get home and think about where I want passflip to head. I should get some kind of test coverage metric thingy going and then approve the PR if it meets a certain expectation. I can't remember what was in here.

**Comment by byxor** [comment]
@devoxel My desire to protect the "integrity" of this codebase was a minor insanity, founded on top of ethereal ideas about the lifetime/complexity/importance of this project.

TL;DR: Sorry for taking so long to accept something so basic. Thanks dude ❤️Hope you're keeping well.

**Review by byxor — COMMENTED** [review]
yo idk how 2 approve but i wanna give it a green tick

**Review by iandioch — COMMENTED** [review]
Green tick

**Comment by devoxel** [comment]
> My desire to protect the "integrity" of this codebase was a minor insanity, founded on top of ethereal ideas about the lifetime/complexity/importance of this project.

I think we could found some kind of new principle on this, puritanical software development. All code checked in must be such beauty that those who look at it wonder not if it functions but how it it can exist in such a cruel world.

But yeah I totally forgot about this PR. I'm very bad for leaving PRs / issues open for several years on Github. All G, I am a nitpicky code reviewer myself, though I do allow people to ignore my awful taste for nitpicks.
